src/vehicle_part.cpp

```diff
@@ -530,6 +530,12 @@ bool vehicle_part::is_tank() const
     return base.is_watertight_container();
 }
 
+bool vehicle_part::contains_liquid() const
+{
+    return is_tank() && !base.contents.empty() &&
+           base.contents.legacy_front().made_of( phase_id::LIQUID );
```

Comment:
```suggestion
           base.contents.only_item().made_of( phase_id::LIQUID );
```

Reply:
OK, I guess this makes sense, since tanks can only hold one kind of liquid.